### PR TITLE
Introduce delay before executing search

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-345](https://trello.com/c/eDPlKMPS/345-development-add-js-superset-to-search)

### Context

Currently the search is executed after every key press, which both results in a lot of search API calls and also means our analytics include search terms for each character.

Instead, we want to introduce a 500ms delay after the user finishes typing before executing the search.

### Changes proposed in this pull request

- Introduce delay before executing search

### Guidance to review

It would have been nicer to use Jest's fake timers for the tests, however the `accessible-autocomplete` lib uses timers to poll the input field internally and this makes using the fake timers not possible (it just hangs, even when you ask it to only run on the pending timers).

Now that there's a delay do we need a 'loading...' entry instead of 'no results' while the user types?

### CODE REVIEW

**Comment by MylesJarvis** [comment]
@ethax-ross I think we need something instead of the 'no results found' that appears. Can we repurpose that to 'searching' or WTTE. I guess we'd then need to ensure 'no results found' continues to surface once the delay has been processed if there are in fact no results

File: app/webpacker/controllers/searchbox_controller.js [hunk 1]
```diff
@@ -82,20 +82,26 @@ export default class extends Controller {
...
+    if (this.delaySearchTimeout) {
+      clearTimeout(this.delaySearchTimeout);
     }
 
-    request.send()
+    this.delaySearchTimeout = setTimeout(() => {
+      this.searchQuery = query
+      this.scheduleSubmissionToAnalytics()
+
+      let request = new XMLHttpRequest()
```
Comment (peteryates) [inline app/webpacker/controllers/searchbox_controller.js:93]: Been a while since I saw a `XMLHttpRequest` in the wild 😂

**Review by peteryates — APPROVED** [review]
Looks good to me.

File: app/webpacker/controllers/searchbox_controller.js (hunk 1 above)
Comment (ethax-ross) [reply app/webpacker/controllers/searchbox_controller.js:96]: hah yeah we should probably swap that out to use `fetch` at some point

**Review by MylesJarvis — APPROVED** [review]
Looks good to me 👍

File: app/webpacker/controllers/searchbox_controller.js (hunk 1 above)
Comment (peteryates) [reply app/webpacker/controllers/searchbox_controller.js:96]: Yeah - sending IE11 users to the non-JS one is fair imo!

**Pull request merged**